app/src/main/java/com/alexP/socialnetwork/presentation/auth/logIn/LogInViewModel.kt

```diff
@@ -0,0 +1,81 @@
+package com.alexP.socialnetwork.presentation.auth.logIn
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.alexP.socialnetwork.presentation.state.RequestState
+import com.alexP.socialnetwork.utils.getValidationResultMessage
+import com.alexp.datastore.DataStoreProvider
+import com.alexp.textvalidation.validateEmail
+import com.alexp.textvalidation.validatePassword
+import com.alexp.webapi.repository.MainRepository
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+
+class LogInViewModel(
+    private val dataStore: DataStoreProvider,
+    private val mainRepository: MainRepository,
+) : ViewModel() {
+
+    private val _requestState = MutableLiveData<RequestState>()
+    val requestState: LiveData<RequestState> = _requestState
+
+    private val _logInFormState = MutableLiveData<LogInFormState>().apply {
+        value = LogInFormState(email = "", password = "")
+    }
+    val logInFormState: LiveData<LogInFormState> = _logInFormState
+
+    fun authorize() {
+        if (_logInFormState.value?.isDataValid == false){
+            updateEmailFormField(_logInFormState.value?.email ?: "")
+            updatePasswordFormField(_logInFormState.value?.password ?: "")
+            return
+        }
+
+        val email = _logInFormState.value?.email ?: return
+        val password = _logInFormState.value?.password ?: return
+        viewModelScope.launch(Dispatchers.IO) {
+            _requestState.postValue(RequestState.Loading)
+            mainRepository.authorize(email, password)
+                .onSuccess { response ->
+                    _requestState.postValue(RequestState.Success)
+                    dataStore.setAccessToken(response.data.accessToken)
+                    dataStore.setRefreshToken(response.data.refreshToken)
+                    dataStore.setUserId(response.data.user.id)
```

Comment:
Actions with datastore does not affect for UI appearance so it can be left somewhere in repository class of business logic.